### ISSUE TITLE
aerospike.shm.use is not working as expected.

### ISSUE THREAD

**Issue by absolute8511** [aerospike/aerospike-client-php#62]
`aerospike.shm.use` seems not work. The `ipcs -m` showed no shm exist.

I digged into the `PHP_METHOD(Aerospike, __construct)`  and the `aerospike_obj_p->as_ref_p->as_p->config`  showed the shm related configures is not updated at all.

**Comment by rbotzer** [comment]
Do you mind stating a few basic things, such as your client version, OS, and its version, your config, the sample script, and the output of:
* `php -i | grep aerospike`
* `ipcs -m`

Thanks.

**Comment by absolute8511** [comment]
```
php -i | grep aerospike:
Additional .ini files parsed => /usr/local/etc/php/5.6/conf.d/conf-aerospike.ini,
aerospike
aerospike support => enabled
aerospike version => 3.4.5
Registered save handlers => files user aerospike
```
```
$ sudo ipcs -m

------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status
0x6c01004d 262144     zabbix     600        527280     6
0x6d01004d 294913     zabbix     600        50648      4
```

the two keys in shm are coming from other services.

**Comment by absolute8511** [comment]
It seems `#define SHM_USE_PHP_INI INI_BOOL("aerospike.shm.use") ? INI_BOOL("aerospike.shm.use") : false`  can not parse true correctly.

Changing to `#define SHM_USE_PHP_INI INI_INT("aerospike.shm.use") ? INI_INT("aerospike.shm.use") : 0` , I can enable shm by `aerospike.shm.use = 1` successfully.

**Comment by rbotzer** [comment]
You definitely stumbled on a typo in the doc (`true` and `false` should not be used). But, you should be able to enable/disable `aerospike.shm` using `1` or `0`.

Using integers works in OS X, Ubuntue 14.04, etc. Are you saying that you can't do this with integer values in your `conf-aerospike.ini`?

For example:
```
$ cat /usr/local/etc/php/5.6/conf.d/aerospike.ini
extension=aerospike.so
aerospike.shm.use=true
aerospike.shm.max_nodes=16
aerospike.shm.max_namespaces=8
aerospike.shm.takeover_threshold_sec=30
$ php -r "var_dump(ini_get('aerospike.shm.use'));"
string(1) "1"
$ php -r "var_dump(ini_get('aerospike.shm.max_nodes'));"
string(2) "16"
```

**Comment by absolute8511** [comment]
I am using the `aerospike.shm.use=true`  as example in document but not working. So I changed to `aerospike.shm.use=1`

**Comment by rbotzer** [comment]
Okay, I get what you're saying. You see the example above (OS X) and here's one from Ubuntu 14.04:
```
$ cat /etc/php5/cli/conf.d/aerospike.ini
extension=aerospike.so
aerospike.udf.lua_system_path=/usr/local/aerospike/lua
aerospike.udf.lua_user_path=/usr/local/aerospike/usr-lua
aerospike.shm.use=true
vagrant@ubuntu-14:~$ php -r "var_dump(ini_get('aerospike.shm.use'));"
string(1) "1"
```

So, which OS are you on?

**Comment by rbotzer** [comment]
By the way, config issue aside (which is minor) we have a pair of issues (internally CLIENT-9 and CLIENT-595) to deal with the shared memory segment's key.

**Comment by absolute8511** [comment]
I tried both on my Mac OSX and on server `CentOS 2.6.32-504.el6.x86_64`

php -r "var_dump(ini_get('aerospike.shm.use'));"
string(1) "1"

**Comment by rbotzer** [comment]
Huh. Okay, I'll look into it more. Which OS X version, by the way?

**Comment by absolute8511** [comment]
It's OS X 10.11.1 (15B42)

**Comment by rbotzer** [comment]
Thanks.

**Comment by rbotzer** [comment]
Fixed in release 3.4.6.
